### PR TITLE
Align queryOptions type in reference components and controller

### LINKED ISSUE #10379: Incorrect type for queryOptions in ReferenceArrayField
**What you were expecting:**
Passing the `queryOptions` prop to `ReferenceArrayField` should not require the `queryKey` property.

**What happened instead:**
Typescript error:

```
Property 'queryKey' is missing in type '{}' but required in type 'UseQueryOptions<RaRecord<Identifier>[], Error, RaRecord<Identifier>[], QueryKey>'.ts(2741)
```

**Steps to reproduce:**
Have a `ReferenceArrayField` component and specify `queryOptions` without `queryKey` for it.

**Related code:**

```ts
<ReferenceArrayField
  queryOptions={{ }}
  reference='resourceName'
  source='referenceIds'>
```

**Other information:**

The component passes `queyOptions` to `useReferenceArrayFieldController`, which uses the correct type by omitting the "forbidden" fields:

```ts
queryOptions?: Omit<
  UseQueryOptions<ReferenceRecordType[]>,
  'queryFn' | 'queryKey'
>;
```

**Environment**

* React-admin version: 5.3.3
* Last version that did not exhibit the issue (if applicable): Unsure, I believe the prop was added ~1 year ago to the component
* React version: 18.3.1
* Browser: Chrome
* Stack trace (in case of a JS error): -

### PR DESCRIPTION
**Pull request by dricholm**
Fixes #10379

## Problem

`ReferenceArrayField`'s `queryOptions` required `queryFn` and `queryKey` props even though these should be optional.

## Solution

Update types to omit the above two fields.
Also aligned the types between `ReferenceField`, `ReferenceArrayField` and `useReferenceFieldController`.

## How To Test

Create a `<ReferenceArrayField queryOptions={{}} />` and confirm no type errors occur on `queryOptions` stating that `queryFn` and `queryKey` are required.

## Additional Checks

- [x] The PR targets `master` for a bugfix, or `next` for a feature
- [ ] The PR includes **unit tests** (if not possible, describe why)
  - Didn't see an existing unit test yet, should a new one be added?
- [ ] The PR includes one or several **stories** (if not possible, describe why)
  - Unsure if story is required for this change. Please let me know if you have an idea.
- [x] The **documentation** is up to date

### CODE REVIEW

File: packages/ra-core/src/controller/field/useReferenceFieldController.ts [hunk 1]
```diff
@@ -52,10 +52,9 @@ export interface UseReferenceFieldControllerOptions<
     ReferenceRecordType extends RaRecord = RaRecord,
 > {
     source: string;
-    queryOptions?: Partial<
-        UseQueryOptions<ReferenceRecordType[], Error> & {
-            meta?: any;
```
[the remark this answers is not in the document]
Comment (dricholm) [reply packages/ra-core/src/controller/field/useReferenceFieldController.ts:57]: Sorry, forgot to mention this in the PR.
`meta` is already present in `UseQueryOptions` and it's also not specified in other parts inside `react-admin`, e.g. `UseGetListOptions` also simply uses `Omit<UseQueryOptions<GetListResult<RecordType>, Error>, 'queryKey' | 'queryFn'>` as base.
To reduce redundancy and also just be more unified I thought removing it here made sense. What do you think?

File: packages/ra-core/src/controller/field/useReferenceFieldController.ts [hunk 2]
```diff
@@ -52,10 +52,9 @@ export interface UseReferenceFieldControllerOptions<
     ReferenceRecordType extends RaRecord = RaRecord,
 > {
     source: string;
-    queryOptions?: Partial<
```
[the remark this answers is not in the document]
Comment (dricholm) [reply packages/ra-core/src/controller/field/useReferenceFieldController.ts:55]: This part didn't show an error, the error was only present in `ReferenceArrayField` as mentioned in the issue. I just decided to unify the type in related places.

[the remark this answers is not in the document]
Comment (fzaninotto) [reply packages/ra-core/src/controller/field/useReferenceFieldController.ts:55]: ah, right, I missed that, thanks for the explanation.

File: packages/ra-core/src/controller/field/useReferenceFieldController.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (fzaninotto) [reply packages/ra-core/src/controller/field/useReferenceFieldController.ts:57]: You're right.

**Comment by fzaninotto** [comment]
Thanks!

**Pull request merged**